src/aggregator/server/rawtcp/server.go

```diff
@@ -118,7 +123,15 @@ func (s *handler) Handle(conn net.Conn) {
 		remoteAddress = remoteAddr.String()
 	}
 
-	reader := bufio.NewReaderSize(conn, s.readBufferSize)
+	connReader := io.Reader(conn)
+	switch s.compressType {
+	case client.SnappyCompressType:
+		connReader = snappy.NewReader(connReader)
+	case client.FlateCompressType:
+		connReader = flate.NewReader(connReader)
+	}
```

Comment:
nit: might be worth adding a `default: //noop` case just for clarity (also in case the linter complains)?